### PR TITLE
[SDPA-3007] Added winston as logger

### PR DESCRIPTION
**Pull request by tim-yao**
**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPA-3007

### Changed

1. Add Winston as logger with levels.
2. Add console and logstash as transport.
3. Integrate with lagoon log so we can search and analyze them in Kibana.
4. Update all console log to Winston logger.

### Screenshots

Screen Shot 2019-08-13 at 12 56 05 pm
Screen Shot 2019-08-13 at 1 03 35 pm

### CODE REVIEW

File: Dockerfile.app [hunk 1]
```diff
@@ -11,10 +11,6 @@ FROM amazeeio/node:10
 COPY --from=builder /app/. /app/
 
 ARG LAGOON_GIT_BRANCH
-ARG CONTENT_API_AUTH_PASS
-ARG CONTENT_API_AUTH_USER
-ENV CONTENT_API_AUTH_PASS ${CONTENT_API_AUTH_PASS}
-ENV CONTENT_API_AUTH_USER ${CONTENT_API_AUTH_USER}
```
Comment (tim-yao) [inline Dockerfile.app]: Remove duplicated lines in Dockerfile, however it's unrelated to this task.

File: packages/ripple-nuxt-tide/lib/server-middleware/logger.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+module.exports = function (req, res, next) {
+  // req is the Node.js http request object
+  const status = res.statusCode
+  const method = req.method.toUpperCase()
+  const reqUrl = decodeURI((url.parse(req.url)).pathname)
+  if (!reqUrl.includes('api/v1/')) {
+    logger.info('Server got request: %s %s %s', status, method, reqUrl, { label: 'Connect' })
+  } else {
+    logger.info('Proxy %s %s to backend, %s.', method, reqUrl, status, { label: 'Connect' })
```
Comment (tim-yao) [inline packages/ripple-nuxt-tide/lib/server-middleware/logger.js]: To log all server connections.

File: packages/ripple-nuxt-tide/lib/core/config-loader.js [hunk 3]
```diff
@@ -210,6 +212,6 @@ export const build = (tideConfig, _this) => {
   // Build custom modules
   buildCustomModules(tideConfig, _this)
   if (process.env.TIDE_DEBUG) {
-    console.info('Tide configuration:', tideConfig)
+    logger.log('debug', 'Tide configuration: %O', tideConfig)
```
Comment (alan-cole) [inline packages/ripple-nuxt-tide/lib/core/config-loader.js]: What's the difference between %s and %0?

File: packages/ripple-nuxt-tide/lib/core/logger.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+// {
+//   error: 0,
+//   warn: 1,
+//   info: 2,
+//   verbose: 3,
+//   debug: 4,
+//   silly: 5
+// }
+
+if (!process.browser) {
```
Comment (alan-cole) [inline packages/ripple-nuxt-tide/lib/core/logger.js]: Should it be process.client?

File: packages/ripple-nuxt-tide/lib/core/logger.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+          return info
+        })(),
+        format.colorize(),
+        printFormat
+      )
+    })
+  ]
+})
+
+if (!process.browser) {
```
Comment (alan-cole) [inline packages/ripple-nuxt-tide/lib/core/logger.js]: is it process.client here as well?

File: packages/ripple-nuxt-tide/lib/core/middleware-helper.js [hunk 6]
```diff
@@ -40,14 +41,14 @@ export const callMiddleware = async (middlewareConfig, context) => {
     // Call other middleware, the order is core modules > custom root > custom modules.
     const middleware = getMiddleware(middlewareConfig)
     if (process.env.TIDE_DEBUG && process.server) {
-      console.info('Tide middleware will be called:', middleware)
+      logger.log('debug', 'Tide middleware will be called: %O', middleware)
```
Comment (alan-cole) [inline packages/ripple-nuxt-tide/lib/core/middleware-helper.js]: I can't seem to find what the %0 represents.

File: packages/ripple-nuxt-tide/modules/webform/conditional-logic.js [hunk 7]
```diff
@@ -24,7 +26,7 @@ function testField (field, data) {
         }
         break
       default:
-        console.warn(`Form: State "${state}" is not supported.`)
+        logger.warn(`Form: State "${state}" is not supported.`, { label: 'Webform' })
```
Comment (alan-cole) [inline packages/ripple-nuxt-tide/modules/webform/conditional-logic.js]: Should we be using the %s syntax for consistency?

File: packages/ripple-nuxt-tide/modules/webform/conditional-logic.js [hunk 8]
```diff
@@ -63,7 +65,7 @@ function prepareTest (rulesObject, data) {
       })
       break
     default:
-      console.warn(`Form: "${rulesType}" rules variable is not supported.`)
+      logger.warn(`Form: "${rulesType}" rules variable is not supported.`, { label: 'Webform' })
```
Comment (alan-cole) [inline packages/ripple-nuxt-tide/modules/webform/conditional-logic.js]: Should we be using the %s syntax for consistency?

File: packages/ripple-nuxt-tide/modules/webform/conditional-logic.js [hunk 9]
```diff
@@ -150,7 +152,7 @@ function performTriggerCheck (rule) {
       result = (rule.modelValue !== rule.triggerValue)
       break
     default:
-      console.warn(`Form: Trigger "${rule.triggerName}" is not supported.`)
+      logger.warn(`Form: Trigger "${rule.triggerName}" is not supported.`, { label: 'Webform' })
```
Comment (alan-cole) [inline packages/ripple-nuxt-tide/modules/webform/conditional-logic.js]: Should we be using the %s syntax for consistency?

File: packages/ripple-nuxt-tide/modules/webform/conditional-logic.js [hunk 10]
```diff
@@ -194,7 +196,7 @@ function performOperatorCheck (operator, results) {
       isPass = (xorCount === 1)
       break
     default:
-      console.warn(`Form: Opeator "${operator}" not supported.`)
+      logger.warn(`Form: Opeator "${operator}" not supported.`, { label: 'Webform' })
```
Comment (alan-cole) [inline packages/ripple-nuxt-tide/modules/webform/conditional-logic.js]: Should we be using the %s syntax for consistency?

File: packages/ripple-nuxt-tide/modules/webform/mapping-filters.js [hunk 11]
```diff
@@ -341,7 +343,7 @@ module.exports = {
         data.schema.groups.push({ 'fields': [field] })
       } else {
         if (process.server) {
-          console.error(new Error(`Webform element type "${element['#type']}" is not supported in nuxt-tide at this stage, please ask site admin to remove it from relative Tide webform or addd support for it.`))
+          logger.warn(`Webform element type "${element['#type']}" is not supported in nuxt-tide at this stage, please ask site admin to remove it from relative Tide webform or addd support for it.`, { label: 'Webform' })
```
Comment (alan-cole) [inline packages/ripple-nuxt-tide/modules/webform/mapping-filters.js]: Should we be using the %s syntax for consistency?

File: packages/ripple-nuxt-tide/lib/core/config-loader.js (hunk 3 above)
Comment (tim-yao) [reply packages/ripple-nuxt-tide/lib/core/config-loader.js]: Winston log format is using node `util.format` You can check the details here: https://nodejs.org/api/util.html#util_util_format_format_args.

Comment (alan-cole) [reply packages/ripple-nuxt-tide/lib/core/config-loader.js]: Cheers

File: packages/ripple-nuxt-tide/lib/core/logger.js (hunk 4 above)
Comment (tim-yao) [reply packages/ripple-nuxt-tide/lib/core/logger.js]: Will change to `process.client`. 👍

File: packages/ripple-nuxt-tide/modules/webform/conditional-logic.js (hunk 7 above)
Comment (tim-yao) [reply packages/ripple-nuxt-tide/modules/webform/conditional-logic.js]: Yes, should. But based on the amount of changes, I feel leave it as it is will make the work done quicker. I will have a fix now anyway!

**Pull request merged**